### PR TITLE
Do not fire pre/post hooks twice on chunk upload

### LINKED ISSUE #23212: Get error by uploading files with activated activity app (Desktop Client)
I get an error when I upload some files (~> 20MB) with the desktop Client (OS X and Windows) and the activity app is activated. If the activity app is disabled everything works fine.

### Steps to reproduce
1. Activate the activity App
2. Upload an file with the owncloud desktop client (OS X / Win)

### Actual behaviour
Error Message: " "no app in context" // "not found while trying to get owner"

### Server configuration
**Operating system**: Ubuntu 14.04. LTS

**Web server:** Apache/2.4.7
 
**Database:** MariaDB 5.5

**PHP version:** PHP5.5

**ownCloud version:**  9.0.0

**ownCloud Client version:**  2.1.1 (build 3107)

**Updated from an older ownCloud or fresh install:** Upgraded from 8.2.3

**Are you using an external user-backend, if yes which one:** no

#### ownCloud log
```
{"reqId":"tvuu3GTNePG4LPGLGEU8","remoteAddr":"{HIDE_IP}","app":"no app in context","message":"Exception: {\"Exception\":\"OCP\\\\Files\\\\NotFoundException\",\"Message\":\"\\\/Bilder\\\/IMG_0015.mov not found while trying to get owner\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(913): OC\\\\Files\\\\View->getOwner('\\\/Bilder\\\/IMG_001...')\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/fileshooks.php(191): OC\\\\Files\\\\Filesystem::getOwner('\\\/Bilder\\\/IMG_001...')\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/fileshooks.php(145): OCA\\\\Activity\\\\FilesHooks->getSourcePathAndOwner('\\\/Bilder\\\/IMG_001...')\\n#3 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/fileshooks.php(101): OCA\\\\Activity\\\\FilesHooks->addNotificationsForFileAction('\\\/Bilder\\\/IMG_001...', 'file_created', 'created_self', 'created_by')\\n#4 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/fileshooksstatic.php(60): OCA\\\\Activity\\\\FilesHooks->fileCreate('\\\/Bilder\\\/IMG_001...')\\n#5 [internal function]: OCA\\\\Activity\\\\FilesHooksStatic::fileCreate(Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/hook.php(105): call_user_func(Array, Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/filechunking.php(224): OC_Hook::emit('OC_Filesystem', 'post_create', Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php(436): OC_FileChunking->file_assemble(Object(OCA\\\\Files_Trashbin\\\\Storage), 'files\\\/Bilder\\\/IM...', '\\\/dominik\\\/files\\\/...')\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php(103): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #49)\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/directory.php(134): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #49)\\n#11 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_0015.mov-ch...', Resource id #49)\\n#12 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(525): Sabre\\\\DAV\\\\Server->createFile('Bilder\\\/IMG_0015...', Resource id #49, NULL)\\n#13 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#14 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#15 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#16 \\\/var\\\/www\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#17 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#18 \\\/var\\\/www\\\/owncloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/ownclo...')\\n#19 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php\",\"Line\":1583}","level":3,"time":"{TIME}"}

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/23212

Let's see what smashbox says...

@nickvergessen @icewind1991 @rullzer @schiesbn

### CODE REVIEW

**Comment by PVince81** [comment]
Seems smashbox was happy with this. Please have a look.

**Comment by nickvergessen** [comment]
Mind fixing the doc block as well?
```php
* @return bool assembled file size or false if file could not be created
```

bool can not be a filesize :scream:

**Comment by PVince81** [comment]
@nickvergessen fixed the doc block and also removed unused param

**Comment by rullzer** [comment]
Uploaded some chunk files. All good.

I like the code cleanup! :+1:

**Comment by PVince81** [comment]
I still wonder whether we should backport this, at least to 9.0.1.
So far no other app was reported broken due to hooks firing twice, but who knows...

**Comment by rullzer** [comment]
@PVince81 I think the backport would be safe to do since it is still the same code. Better to backport it now then when somebody writes an app that breaks because of it later.

File: lib/private/filechunking.php [hunk 1]
```diff
@@ -179,59 +179,22 @@ public function signature_split($orgfile, $input) {
 	 * Assembles the chunks into the file specified by the path.
 	 * Also triggers the relevant hooks and proxies.
 	 *
-	 * @param \OC\Files\Storage\Storage $storage
+	 * @param \OC\Files\Storage\Storage $storage storage
 	 * @param string $path target path relative to the storage
-	 * @param string $absolutePath
-	 * @return bool assembled file size or false if file could not be created
+	 * @param string $absolutePath absolute path
```
Comment (nickvergessen) [inline lib/private/filechunking.php]: param was removed

Comment (PVince81) [inline lib/private/filechunking.php]: fixed

File: lib/private/filechunking.php [hunk 2]
```diff
@@ -179,59 +179,21 @@ public function signature_split($orgfile, $input) {
 	 * Assembles the chunks into the file specified by the path.
 	 * Also triggers the relevant hooks and proxies.
 	 *
-	 * @param \OC\Files\Storage\Storage $storage
+	 * @param \OC\Files\Storage\Storage $storage storage
 	 * @param string $path target path relative to the storage
-	 * @param string $absolutePath
-	 * @return bool assembled file size or false if file could not be created
+	 * @return int|bool assembled file size or false if file could not be created
```
Comment (nickvergessen) [inline lib/private/filechunking.php]: It always is bool due to `return $count > 0;` ?!

Comment (PVince81) [inline lib/private/filechunking.php]: Ah... dammit... I looked at this several times and saw `$count` and not `> 0` :tired_face:

**Comment by PVince81** [comment]
Fixed and rebased again

**Comment by icewind1991** [comment]
Code looks good :+1:

**Pull request merged**